### PR TITLE
Terminate the engine gracefully

### PR DESCRIPTION
**Pull request by bzsolt**
IoT.js-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by bzsolt** [comment]
https://github.com/Samsung/iotjs/pull/1142 fixes timer leak.

File: src/iotjs.c [hunk 1]
```diff
@@ -66,6 +66,10 @@ static bool iotjs_jerry_initialize(const iotjs_environment_t* env) {
   // Set magic strings.
   iotjs_register_jerry_magic_string();
 
+  // Register VM execution stop callback.
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_environment_t, env);
+  jerry_set_vm_exec_stop_callback(vm_exec_stop_callback, &(_this->state), 6);
```
Comment (daeyeon) [inline src/iotjs.c]: IMHO, we can handle exceptions in iotjs_uncaught_exception. Do we truely need to register this callback? Or, do you have any plan to use it for a certain purpose?

Comment (bzsolt) [reply src/iotjs.c]: Yes we do, this callback is required for the vm-exec-stop feature of JerryScript, and allow to terminate the script execution when we want to abort that.
You can find a more detailed description about this feature [here](http://jerryscript.net/api-reference/#jerry_set_vm_exec_stop_callback).
(I've added a more verbose commit message too.)

Comment (daeyeon) [reply src/iotjs.c]: @bzsolt Thanks for your explanation. One more thing, do you believe that 6 for `frequncy` is appropriate?

Comment (bzsolt) [reply src/iotjs.c]: Previously I did some experiment and this value seemed to be enough, but now I found a testcase which requires to decrease the frequency to 2. Later we could fine-tune this value further if needed.

**Comment by daeyeon** [comment]
LGTM. I love this patch.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**